### PR TITLE
Email admin(s) when a new application is submitted

### PR DESCRIPTION
**Pull request by weedySeaDragon**
**PT Story: https://www.pivotaltracker.com/story/show/149374145**

### Changes proposed in this pull request:
1.  send all admins an email when a new membership application is submitted 
2.  added  `admins` scope for User so that we can get all admins and send email to them
3.  added details to the email so that the admin can see the number of files uploaded for the application.  (Recently a user said she uploaded files but they weren't in SHF.  Providing this info in the email to the admins will help admins to know if the files were successfully uploaded and saved in SHF or not.)

### Screenshots (Optional):

**text version:**

shf-admin-new-app-txt-2

**html version:**

shf-admin-new-app-html-2

### Ready for review:
@AgileVentures/shf-project-team

### CODE REVIEW

**Review by thesuss — APPROVED** [review]

File: app/mailers/admin_mailer.rb [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
+    mail to: @recipient_email, subject: t('application_mailer.admin.new_application_received.subject')
 
+  end
 
 
   private
 
+  def set_mail_info(method_sym, admin)
+    @greeting_name = set_greeting_name admin
+    @recipient_email = set_recipient_email admin
```
Comment (patmbolger) [inline app/mailers/admin_mailer.rb]: Don't need these assignments as the utility methods already set the instance var, i.e. these statements should be:
```
set_greeting_name admin
set_recipient_email admin
```

**Review by patmbolger — CHANGES_REQUESTED** [review]
Looks good.  Tested and worked for me.

See one suggested change.

File: app/mailers/admin_mailer.rb (hunk 1 above)
Comment (weedySeaDragon) [reply app/mailers/admin_mailer.rb]: good catch!

**Comment by weedySeaDragon** [comment]
@thesuss Should we add some wording in this email to point out that payment is still needed?
(I thought of this after your comment in #416 ).

**Review by patmbolger — APPROVED** [review]

File: app/controllers/membership_applications_controller.rb [hunk 2]
```diff
@@ -49,15 +49,16 @@ def create
     if @membership_application.save
 
       if new_file_uploaded(params)
+
+        MembershipApplicationMailer.acknowledge_received(@membership_application).deliver
```
Comment (RobertCram) [inline app/controllers/membership_applications_controller.rb]: It seems `deliver` is deprecated (?), so you might want to change this to `deliver_now`

File: app/controllers/membership_applications_controller.rb [hunk 3]
```diff
@@ -49,15 +49,16 @@ def create
     if @membership_application.save
 
       if new_file_uploaded(params)
```
Comment (RobertCram) [inline app/controllers/membership_applications_controller.rb]: There is a problem here that is not directly related to this PR, but should be addressed: If one of the files is too big or has an unacceptable extension, the application will be saved but the email message will never be send (`new_file_uploaded` fails). If the files are added in a second attempt, the action has now become an edit operation, and again no message will be send.

I think the easiest fix would be to move the code that sends the mail out of the `new_file_uploaded` conditional and place it after. If the e-mail could also list the names of the uploaded files that would serve as a great check for the user.

**Review by RobertCram — CHANGES_REQUESTED** [review]

**Comment by thesuss** [comment]
Wrong place for this comment. The text here is fine.

**Comment by weedySeaDragon** [comment]
@thesuss :  I added the names of the files uploaded.  Is this OK?

**Comment by thesuss** [comment]
@weedySeaDragon 
File-names are usually irrelevant. Most come from FB and have names like "24312415_905701216265640_3935260062927433455_n" I think that might trigger spam flags in the e-mail?

**Review by RobertCram — APPROVED** [review]

**Pull request merged**